bin/tootctl

```diff
@@ -5,9 +5,15 @@ require_relative '../config/boot'
 require_relative '../lib/mastodon/cli/main'
 
 begin
+  ENV['THOR_DEBUG'] = '1' # Enables the script to rescue `Thor::Error`
```

Comment:
I agree it's weird. I found a pretty old discussion where the bundler script had solved the same issue the same way.

I just looked at this again and I think there's a `debug` option we can pass it to force the error to raise, which is probably better. Will attempt them and see how it goes.

Reply:
Yes - turns out newer bundler/thor since the discussion I found are just using the debug option directly instead of this workaround. Updated the PR to do that, preserving comment.

Reply:
That still feels a bit weird to me—why is it “debug” and why do we have to override a possibly user-set option?—but definitely cleaner than an env variable.

Reply:
The default thor behavior (debug=false) is to catch `Thor::Error`, print the error message, and then either `exit(false)` or continue execution, depending on the `exit_on_failure?` setting (we set to true in our base class).

So, in order to preserve the "print out the error message AND make it red" behavior of the previous script, I needed to override this failure/exit behavior. I'd have to double-check this, but I think if we didn't care about the red color portion, we could in fact just raise this error class with our messages and it would print the message and exit non-zero, which is what we want.

Separately, like you implied, we could add a new error class to our CLI code, have this top-level script handle that as a special case, and leave the Thor::Error alone with its default behavior.

Reply:
Oh, to answer your actual question! -- with debug=true thor will just re-raise the error and continue on w/out exiting, which is why this was necessary here.

I think in this scenario, the tootctl script effectively IS the user of thor, and it's fine to do this way ... though I guess it's possible someone would want more control over that? The code does check both the env var and the option, so I think you could still override via env var. While reading about this, I found many blog posts with an observation like "this is very odd default behavior from thor", which I sort of agree with.

Reply:
Looking at Thor's source code, this behavior (re-raising `Thor::Error`) is the only thing affected by `THOR_DEBUG`/the debug option, so I guess this is an ok change.

Reply:
Yeah - I went down that path a bit ... there is a public `say_error` method on the shell classes which we could probably configure with color by using that shell class, but this debug/catching section uses a seperate internal-only error method to print the error out. It did not seem easily configurable when I looked.

Reply:
I think you can pass a `shell` option, which has an `error` method used exclusively for that. But we can have a look in a different PR.